leader/src/rpc.rs

```diff
@@ -0,0 +1,199 @@
+use anyhow::Context;
+use ethers::types::{Address, Bloom, H256, U256};
+use evm_arithmetization::proof::BlockHashes;
+use futures::{stream::FuturesOrdered, TryStreamExt};
+use reqwest::IntoUrl;
+use serde::Deserialize;
+use tracing::info;
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct EthGetBlockByNumberResult {
```

Comment:
I don't think this is ever used?

Reply:
Hmm it should be. It's accessed in `fetch_previous_block_hashes`.